### PR TITLE
Add reproducible build check step in the proces of Provider releases

### PR DESCRIPTION
**Pull request by potiuk**
The reproducible build we just added makes it possible to run full reproducible package builds (byte-to-byte reproducible) - which means that whoever uses the same sources of Airflow (and official 3rd-party build tools) should get identical, byte-to-byte reproducible packages.

This PR updates process of Provider package verification to make sure the packages are binary-identical, thus removing the need of verifying the sources included in the package (allowing to verify not only the sources coming from the repository, but also the generated content (such as get_provider_info.py files).

One step closer to SLSA compliance of our build/release process.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — APPROVED** [review]
Really nice

**Comment by potiuk** [comment]
> Really nice

Indeeed. Those "reproducible builds" - once we have good tooling - are really powerful in making "secure" supply chain possible

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 1]
```diff
@@ -667,6 +667,102 @@ docker run --rm --entrypoint "airflow" local/airflow info
...
 ```
 
+### Reproducible package builds checks
+
+For provider packages we introduced a reproducible build mechanism - which means that whoever wants
+to use sources of Airflow from the release tag, can reproducibly build the same "wheel" and "sdist"
+packages as the release manager and they will be byte-by-byte identical, which makes them easy to
+verify - if they came from the same sources. This build is only done using released dependencies
+from PyPI and source code in our repository - no other binary dependencies are used during the build
+process and if the packages produced are byte-by-byte identical with one that we can do from tagged sources
```
Comment (hussein-awala) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
process and if the packages produced are byte-by-byte identical with the one we create from tagged sources
```
Or
```suggestion
process and if the packages produced are byte-by-byte identical with the one we build from tagged sources
```

**Review by hussein-awala — APPROVED** [review]
👍

**Pull request merged**